lib/features/map_view/widgets/map_widget.dart

```diff
@@ -38,30 +22,40 @@ class _MapWidget<T extends GoogleNavigable> extends ConsumerWidget {
     final activeItem = ref.watch(context.activeMarkerController<T>());
 
     final locationStatus = ref.watch(locationPermissionStatusProvider);
+
     return switch (locationStatus) {
       AsyncLoading() => const SizedBox(),
       AsyncError(:final error) => MyErrorWidget(error),
-      AsyncValue(:final value) => GoogleMap(
-          initialCameraPosition: MapWidgetConfig.defaultCameraPosition,
-          onMapCreated: mapController.onMapCreated,
-          onTap: mapController.onMapBackgroundTap,
-          markers: items
-              .map(
-                (item) => context.markerBuilder<T>()(
-                  item,
-                  ref,
-                  isActive: activeItem == item,
-                ),
-              )
-              .toSet(),
-          myLocationEnabled: value ?? false,
-          myLocationButtonEnabled: value ?? false,
-          mapToolbarEnabled: false,
-          zoomControlsEnabled: false,
-          padding: EdgeInsets.only(
-            top: MediaQuery.paddingOf(context).top,
-            bottom: ref.watch(bottomSheetPixelsProvider),
-          ),
+      AsyncValue(:final value) => Consumer(
+          builder: (BuildContext context, WidgetRef ref, Widget? child) {
+            final asyncItems = ref
+                .watch(context.mapSourceRepository<T>())
+                .value
+                .whereNonNull
+                .toList();
```

Comment:
I wonder if we can use here immutable list? (`IList`)

btw the way `Consumer `here's introduced improved visibility a lot imo.

Reply:
Btw, is this a change that reduced flickering? Moving state refreshment inside `AsyncValue`?

Reply:
Yeah we can use IList here

I moved the state listening down the tree in few places, meaning less widgets are refreshed needlesly. But it still doesn't help :P with the flickering